Alerting: grafana managed group names

gillesdemey opened the pull request:
**What this PR does / why we need it**:

This PR introduces proper support for arbitrary group names in Grafana managed alerts. Previously all Grafana managed rules were placed in to their own group of exactly one rule.

In an effort to unify the behavior between Prometheus rules and Grafana managed rules and to comply with the [Prometheus Alert Generator Compliance Specification](https://github.com/prometheus/compliance/blob/main/alert_generator/specification.md) this PR adds the following:

### An additional tab to view Grafana managed rules in their group hierarchy

The existing grouped view has been renamed to "List" since it flattens the view hierarchy for Grafana managed rules.
A new tab has been added called "grouped"

<img width="516" alt="image" src="https://user-images.githubusercontent.com/868844/162472405-a87ed085-a15c-49a7-bb60-6b1c66734ef1.png">

<img width="576" alt="image" src="https://user-images.githubusercontent.com/868844/162473025-0d590cd6-67c4-4660-9e7b-73a40bd64f70.png">

### Rules can be moved to other groups

In addition to moving rules between namespaces you can now also move rules to another group within the same namespace. The existing `saveGrafanaRule` has been substantially refactored.

### Additional information is included

To better communicate the new alerting paradigm I've included some inline documentation on folder and group. I'll need some input from @achatterjee-grafana so we can refine the content.

<img width="727" alt="image" src="https://user-images.githubusercontent.com/868844/162473341-9ac0df7d-9bba-4309-a8b1-4af6c97aad94.png">

Since the rule interval will also determine the interval of all other rules in the group we have to explicitly call this out.

<img width="528" alt="image" src="https://user-images.githubusercontent.com/868844/162473459-506f782f-c4a3-433a-b625-d9f6aa2b2e10.png">

I'm not quite sure if the interval configuration is in the right place – this is something I'd like @jessover9000's opinion on.

**Which issue(s) this PR fixes**:

Related to #45881

**Special notes for your reviewer**:

I still have to do a quick review myself as I've touch on quite a few files and probably have some TODOs sprinkled around and some variable and function names could be improved – feel free to help bikeshed :)

gillesdemey committed:
chore: transform grafana managed rules response

gillesdemey committed:
initial support for arbitrary group names in grafana managed rule

gillesdemey committed:
lint

gillesdemey committed:
feat: add separate groups view

gillesdemey committed:
handle moving rules to different namespace or group

gillesdemey committed:
rewrite "saveGrafanaRule"

we now support arbitrary group names

* updating an existing rule
* moving a rule to a different group within the same namespace
* moving a rule to a different namespace
* creating a new rule in any namespace and group

gillesdemey committed:
better error

gillesdemey committed:
make group name required

gillesdemey committed:
refactor saveGrafanaRule

makes it a bit easier to reason about

gillesdemey committed:
redirect back to alerts list if uid is empty

we can't figure out the uid of an alerts when it was moved to another group or namespace

gillesdemey committed:
adds field hints

gillesdemey committed:
clarify that eval interval applies to all rules within the group

gillesdemey committed:
clarify the impact of interval

gillesdemey committed:
cleanup

gillesdemey committed:
fix group interval update / create

papagian committed:
Release engineering squad owns `pkg/build` (#47546)

alexandrst88 committed:
bugfix: fix proper type of setting maxConcurrentShardRequests (#47120)

gillesdemey committed:
Merge remote-tracking branch 'origin' into feat/grafana-managed-group-names

gillesdemey committed:
fix test suites

marefr committed:
Instrumentation: Proxy status code correction and various improvements (#47473)

For a proxied request, e.g. Grafana's datasource or plugin proxy:
If the request is cancelled, e.g. from the browser, the HTTP status code is 
now 499 Client closed request instead of 502 Bad gateway.
If the request times out, e.g. takes longer time than allowed, the HTTP status 
code is now 504 Gateway timeout instead of 502 Bad gateway.
This also means that request metrics and logs will get their status codes 
adjusted according to above.

Fixes #46337
Fixes #46338

IevaVasiljeva committed:
Access control: SQL filtering for annotation listing (#47467)

* pass in user to attribute scope resolver

* add SQL filter to annotation listing

* check annotation FGAC permissions before exposing them for commenting

* remove the requirement to be able to list all annotations from annotation listing endpoint

* adding tests for annotation listing

* remove changes that got moved to a different PR

* unused var

* Update pkg/services/sqlstore/annotation.go

Co-authored-by: Ezequiel Victorero <evictorero@gmail.com>

* remove unneeded check

* remove unneeded check

* undo accidental change

* undo accidental change

* doc update

* move tests

* redo the approach for passing the user in for scope resolution

* accidental change

* cleanup

* error handling

Co-authored-by: Ezequiel Victorero <evictorero@gmail.com>

mildwonkey committed:
chore: remove golang.org/x/net/context in favor of stdlib (#47532)

This PR removes golang.org context imports under pkg/services/* and replaces them with the stdlib context.

Closes #44178

eleijonmarck committed:
Add docs for service accounts 8.5 (#46801)

* initial doc for service accounts

* service account token calls complete

* service account tasks

* Update docs/sources/http_api/serviceaccount.md

* adding a token to the service account

* removed unused file

* refactor: review comments

* feat: add API key documentation

* fix: spelling

* Update docs/sources/administration/service-accounts/about-service-accounts.md

Co-authored-by: Jguer <joao.guerreiro@grafana.com>

* Update docs/sources/administration/service-accounts/about-service-accounts.md

Co-authored-by: Jguer <joao.guerreiro@grafana.com>

* Update docs/sources/http_api/serviceaccount.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Update docs/sources/administration/service-accounts/enable-service-accounts.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Update docs/sources/administration/service-accounts/enable-service-accounts.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Update docs/sources/administration/service-accounts/enable-service-accounts.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Update docs/sources/administration/service-accounts/enable-service-accounts.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Update docs/sources/administration/service-accounts/enable-service-accounts.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Update docs/sources/administration/api-keys/about-api-keys.md

* refactor: based on review

* removed the permissions for apikeys, as they are not necessary

* Apply suggestions from code review

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Update docs/sources/administration/service-accounts/create-service-account.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* update based on review

* Fix formatting of bullet points

* formatting

* refcator

Co-authored-by: Jguer <joao.guerreiro@grafana.com>
Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>
Co-authored-by: Vardan Torosyan <vardants@gmail.com>

yuri-tceretian committed:
Alerting: reduce database calls in prometheus-comptible rules API (#47080)

* move validation at the beginning of method
* remove usage of GetOrgRuleGroups because it is not necessary. All information is already available in memory.
* remove unused method

connorlindsey committed:
Tracing: Add README files for tracing datasource plugins

ashharrison90 committed:
Navigation: wrap long items in the menu correctly (#47431)

* start implementing wrapping behaviour

* handle more cases

* minWidth > width to preserve mobile behaviour

* Better css

ying-jeanne committed:
add feature toggle with new format (#47336)

* add feature toggle with new format

* fix some comments :heart:

* Update pkg/infra/log/term/terminal_logger.go

Co-authored-by: Emil Tullstedt <emil.tullstedt@grafana.com>

Co-authored-by: Emil Tullstedt <emil.tullstedt@grafana.com>

mildwonkey committed:
chore: remove all remaining uses of  golang.org/x/net/context (#47564)

* chore: remove all remaining uses of  golang.org/x/net/context

This PR finishes the work started in #47532, replacing all calls to that package with the stdlib context and using http.NewRequestWithContext to include the context where necessary.

Bonus: small formatting fixes to goimports in these files.

closes #44178

* tweak: use context.Background in favor of TODO for tests

james-deee committed:
CloudWatch: Added missing MemoryDB Namespace metrics (#47290)

CloudWatch: Added missing MemoryDB metrics

yuri-tceretian committed:
Alerting: Add check for datasource permission in alert rule read API (#47087)

* add check for access to rule's data source in GET APIs

* use more general method GetAlertRules instead of GetNamespaceAlertRules.
* remove unused GetNamespaceAlertRules.

Tests:
* create a method to generate permissions for rules
* extract method to create RuleSrv
* add tests for RouteGetNamespaceRulesConfig

Commit:
Azure Monitor: Bug Fix for for template variables (#47478)

joeblubaugh committed:
Unified Alerting: UI explains "match all" case in Notification Policies.  (#47574)

This change makes it explicit in the UI that a notification policy with no matching labels matches all alerts that it processes. There are visual changes in both the Notification Policy editor, and in the Notification Policy table where matching columns are shown.

It's valid to have a notification policy with no label matchers attached. Such a policy matches all alerts sent to it. It's a common stumbling block for users. Where are all my alerts going?

Co-authored-by: gillesdemey <gilles.de.mey@gmail.com>

gguillotte-grafana committed:
Explore/Logs: Improve and add descriptions of Explore's Escape newlines feature (#46709)

* Explore/Logs: Clarify phrasing of newline escape fix tooltip

Rewrite the tooltip for the smart feature in PR #31352 that
replaces incorrectly escaped newlines in log lines.

-   Clarify the functionality of the feature to emphasize its
    interactivity.
-   Remove language suggesting that the feature is experimental,
    when we intended to suggest reviewing the results manually for
    correctness.

* Docs: Document escape newlines feature

Describe and provide steps for using the "Escape newlines" feature
in Explore added in PR #31352.

* Rewrite topic lead to clarify conditional behavior

* Describe reversion of replacements as a standalone task

Reverting the replacements is a separate action, so it should have
its own task. This also describes the option's transformation and
tightens the task language.

* Remove Grafana 7 version qualifier

* Clarify escape sequence detection lede on task

Co-authored-by: Fiona Artiaga <89225282+GrafanaWriter@users.noreply.github.com>

* Clarify "Remove escaping" state change

Co-authored-by: Fiona Artiaga <89225282+GrafanaWriter@users.noreply.github.com>

* Clarify confidence of tooltip content

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

Co-authored-by: Fiona Artiaga <89225282+GrafanaWriter@users.noreply.github.com>
Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

ying-jeanne committed:
Fix unmaarshal of double pointer (#47586)

* Fix unmaarshal of double pointer

* update sdk version

gamab committed:
AccessControl: Make the built-in role definitions public (#47525)

* AccessControl: Make the built-in role definitions public

* Add context to RegisterFixedRoles

Co-authored-by: Jguer <joao.guerreiro@grafana.com>

* Making BuiltInRolesWithParents public to the AccessControl package

Co-authored-by: Jguer <joao.guerreiro@grafana.com>

Co-authored-by: Jguer <joao.guerreiro@grafana.com>

dsotirakis committed:
Storybook: Add `store-storybook` command (#47557)

* Add store-storybook command

* Update grabpl version

peterholmberg commented on public/app/features/alerting/unified/utils/rulerClient.ts:
```diff
@@ -157,65 +141,106 @@ export function getRulerClient(rulerConfig: RulerDataSourceConfig): RulerClient
-    // set group name to rule name, but be super paranoid and check that this group does not already exist
-    const groupName = getUniqueGroupName(values.name, targetFolderGroups);
-    formRule.grafana_alert.title = groupName;
+  // add a rule to a namespace and group
```
maybe this comment is ambiguous 🙂

gillesdemey replied:
Definitely, will remove

peterholmberg commented on public/app/features/alerting/unified/utils/rulerClient.ts:
```diff
@@ -157,65 +141,106 @@ export function getRulerClient(rulerConfig: RulerDataSourceConfig): RulerClient
+    return { uid: '', ruleSourceName: GRAFANA_RULES_SOURCE_NAME };
+  };
+
+  // we can't move the rule atomically so we have to
```
nit: spelling

gillesdemey replied:
I meant so clarify here that there isn't a single atomic operation we can perform (atomically), but I can change the wording :D

peterholmberg reviewed (commented):
Just some minor nits. But I have two questions.

1. I see "default" mentioned in some places. Is this referring to a default folder that we talked about?
2. How are current Grafana managed rules migrated to fit within a group?

ivanahuckova committed:
Query history: Save user preferences in database (#47506)

gabor committed:
loki: use single-dataframe format on the backend (#47069)

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 8.4.6 (#47612)

* ReleaseNotes: Updated changelog and release notes for 8.4.6

* update changelog and release notes

Co-authored-by: Agnès Toulet <agnes.toulet@gmail.com>

gabor committed:
loki: removed unused fields, params, attributes (#47571)

* loki: refactor: removed unused param

* loki: removed unused attribute

* loki: removed unused fields

* loki: remove unused query.format

* loki: removed unused prop

felixdoerre committed:
fix: postgresql ipv6 host should not be enclosed in square brackets (#47522)

IevaVasiljeva committed:
Access-control: documentation for general folder uid restrictions (#47567)

* add caveats for general folder

* expand the list of resources covered by fgac

* Update folder.md

remove unneeded comment

gabor committed:
loki: backend-mode: allow for dashboards too (#47578)

gabor committed:
Loki: use millisecond-steps (#47572)

* loki: use millisecond-steps

* better comment

Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>

* better comment

Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>

gillesdemey committed:
chore: remove redundant code comment [skip ci]

gillesdemey committed:
chore: clarify code comment [skip ci]

torkelo committed:
Explore: Allow users to save Explore queries to dashboards  (#47083)

* Select: Expose AsyncSelectProps interface

* DashboardPicker: Add a generic DashboardPicker component

* Dashboard Service: improve types

* Explore: allow saving explore state in a new panel in an existing dashboard

* Handle saving provisioned dashboards error

* Improve test coverage

* simplify test setup

* Strip base path from url when redirecting to a dashboard

* Keep existing variables when saving to an existing dashboard

* group assertions in test

* SearchCard: handle undefined in meta.updated

* Change required error message

* Add to dashboard alternative

* Add to existing is working

* Add to dashboard form

* remove default add-panel when creating a dashboard from explore

* types cleanup

* remove unneeded BE change

* simplify selector

* Add explore2Dashboard feature toggle

* add tests

* Small refactor & add tests

* small DashboardPicker improvements

* use partial from lodash

* Better error handling

* improve tests & disable button when there are no queries

* rename addPanelToDashboard function

* remove localStorage item if opening tab fails

* UI touchups & tracking

* Fix tests & remove close reporting

* remove echologger debug

* fix adding a panel to an existing dashboard

* Enable explore2Dashboard by default and add docs

* Ensure each panel in dashboards has a valid ID

* force CI restart

Co-authored-by: Elfo404 <me@giordanoricci.com>

AgnesToulet committed:
Chore: update latest.json (#47627)

dprokop committed:
Allow queries import when changing data source type (#47435)

* Enable queries import when changing datasource

* Supporting empty imports

* Review applied

konrad147 commented on public/app/features/alerting/unified/components/rules/RulesTable.tsx:
```diff
@@ -137,8 +136,18 @@ function useColumns(showSummaryColumn: boolean, showGroupColumn: boolean) {
-          return isCloudRulesSource(rulesSource) ? `${namespace.name} > ${group.name}` : namespace.name;
+
+          // ungrouped rules are rules that are in the "default" group name
+          const isUngrouped = group.name === 'default';
```
This code looks almost the same as https://github.com/grafana/grafana/blob/81a5df591caa1fe0097b34e7b142297b3e50f750/public/app/features/alerting/unified/components/rules/RulesGroup.tsx#L122:L129
Maybe we can extract a common component for them?

gillesdemey replied:
Good idea!

ying-jeanne committed:
Linter: fix linter for golangci-lint version v1.45.0 (#47623)

* Update golangci-lint to v1.45.2

Version 1.45.0 added support for Go 1.18:

https://github.com/golangci/golangci-lint/blob/master/CHANGELOG.md#v1450

Signed-off-by: Marcelo E. Magallon <marcelo.magallon@grafana.com>

* Linter: fix some lints issue for golangci-lint version 1.45.2

Co-authored-by: Marcelo E. Magallon <marcelo.magallon@grafana.com>
Co-authored-by: Gábor Farkas <gabor.farkas@gmail.com>

konrad147 commented on public/app/features/alerting/unified/utils/rulerClient.ts:
```diff
@@ -157,65 +141,105 @@ export function getRulerClient(rulerConfig: RulerDataSourceConfig): RulerClient
+    const newRules = existingRule.group.rules
+      .filter((rule): rule is RulerGrafanaRuleDTO => isGrafanaRulerRule(rule))
+      .filter((rule) => rule.grafana_alert.uid !== uid)
+      .concat(newRule as RulerGrafanaRuleDTO); // TODO why type cast this?
```
TODO

gillesdemey replied:
It was being typecast before, I didn't dig too deep in to the reason why – might dig a little deeper if I have the time.

konrad147 commented on public/app/features/alerting/unified/utils/rulerClient.ts:
```diff
@@ -90,13 +79,8 @@ export function getRulerClient(rulerConfig: RulerDataSourceConfig): RulerClient
-      await deleteRulerRulesGroup(rulerConfig, namespace, group.name);
-      return;
-    }
-    // in case of CLOUD
```
Nice refactoring in this file 🚀

konrad147 commented on public/app/features/alerting/unified/components/rules/RulesFilter.tsx:
```diff
@@ -13,10 +13,15 @@ import { Stack } from '@grafana/experimental';
-    label: 'Groups',
+    icon: 'list-ul',
+    label: 'List',
     value: 'group',
```
Can we change the `group` to `list`? Or we should keep it due to compatibility reasons?

aocenas committed:
Prometheus: Fix executing query onBlur in Explore (#47561)

kaydelaney committed:
Dashboard: Refactor panel cleanup (#47323)

axelavargas committed:
Navigation: Show only + icons in overlay menu for new NavBar (#47347)

* Nav: Show overlay icons based on allowed list

* user essentials mob! :trident:

* Navigation: clean up and use new backend prop to show plus icons and
improve visual styling

* Nav: Fix top padding

* refactor to not use showIconInNavbar in NavBarMenuItem

* remove a missed bit

* refactor icon into const

Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>

aocenas committed:
Prometheus: Move count_values from function to aggregation (#47260)

* Move count_values from function to aggregation

* Fix typos

* Fix loki operations

* Fix error that change the aggregation variant on blur

* Fix loki ops

andresmgot committed:
AzureMonitor: Skip App Insights check for v9 (#47624)

fridgepoet committed:
Change to pointer receiver in mock call (#47575)

ryantxu committed:
PanelQueryRunner: use refId from results if the `key` value was not set in the packet (#47598)

dsotirakis committed:
CI: Remove unused downstream build pipelines (#47646)

* Remove unused downstream build pipelines

* Prettify

andresmgot committed:
AzureMonitor: Fix Kusto editor schema (#47576)

mildwonkey committed:
chore: go mod tidy (#47648)

ifrost committed:
Temporarily skip a test that is intermittently timing out and causing build failures. (#47637)

ArturWierzbicki committed:
Storage: remove pointer on write path (#47580)

* remove pointer to contents in UpsertFileCommand

* add command to UpsertFileCommand

* fix client after API refactor

gabor committed:
Loki: improve backend-compatibility (#47579)

* loki: annotation queries: make them ready for the backend-mode

* removed unused code

IevaVasiljeva committed:
Annotation FGAC checks for comments (#47468)

* typo

* remove unwanted change

* remove unwanted change

ifrost committed:
Query History: Move local storage specific settings to persistence layer (#47500)

* Load Rich History when the container is opened

* Store rich history for each pane separately

* Do not update currently opened query history when an item is added

It's impossible to figure out if the item should be added or not, because filters are applied in the backend. We don't want to replicate that filtering logic in frontend. One way to make it work could be by refreshing both panes.

* Test starring and deleting query history items when both panes are open

* Remove e2e dependency on ExploreId

* Fix unit test

* Assert exact queries

* Simplify test

* Fix e2e tests

* Fix toolbar a11y

* Reload the history after an item is added

* Fix unit test

* Remove references to Explore from generic PageToolbar component

* Update test name

* Fix test assertion

* Add issue item to TODO

* Improve test assertion

* Simplify test setup

* Move query history settings to persistence layer

* Fix test import

* Fix unit test

* Fix unit test

* Test local storage settings API

* Code formatting

* Fix linting errors

* Add an integration test

* Add missing aria role

* Fix a11y issues

* Fix a11y issues

* Use divs instead of ul/li

Otherwis,e pa11y-ci reports the error below claiming there are no children with role=tab:

Certain ARIA roles must contain particular children
   (https://dequeuniversity.com/rules/axe/4.3/aria-required-children?application=axeAPI)

   (#reactRoot > div > main > div:nth-child(3) > div > div:nth-child(1) > div >
   div:nth-child(1) > div > div > nav > div:nth-child(2) > ul)

   <ul class="css-af3vye" role="tablist"><li class="css-1ciwanz"><a href...</ul>

* Clean up settings tab

* Remove redundant aria label

* Remove redundant container

* Clean up test assertions

ArturWierzbicki committed:
Previews: create crawler auth setup service (#47349)

* #46968: add `RetrieveServiceAccountIdByName` to serviceaccounts service

* #46968: improve error logging in rendering service

* #46968: add oss crawler account setup

* #46968: fix tests

* #46968: switch back to ROLE_ADMIN

* #46968: rename to crawlerAuth

* comment crawler_auth.go

torkelo committed:
Performance: Create separate div for portal root to limit reflow -> recalc style (#47633)

* Performance: Create separate div for portal root to limit reflow -> recalc style

* refactoring

konrad147 commented:
I wonder if we also should unify the way we allow to edit rule groups? 
For cloud rules, the edit button opens the rule group's edit modal
![image](https://user-images.githubusercontent.com/6293563/163121942-268ec298-2d79-4777-91e8-3ec3f94af150.png)
Whereas, for Grafana Managed rules the same button opens the folder edit view. Moreover, this rule group edit modal gives us a better place to set the evaluation interval.
I think we don't need to do this in this PR. It's just something we may want to consider doing in the future

jackw committed:
Chore: Bump minimist to 1.2.6 (#47401)

dependabot[bot] committed:
Bump go.opentelemetry.io/otel/trace from 1.2.0 to 1.6.3 (#47544)

Bumps [go.opentelemetry.io/otel/trace](https://github.com/open-telemetry/opentelemetry-go) from 1.2.0 to 1.6.3.
- [Release notes](https://github.com/open-telemetry/opentelemetry-go/releases)
- [Changelog](https://github.com/open-telemetry/opentelemetry-go/blob/main/CHANGELOG.md)
- [Commits](https://github.com/open-telemetry/opentelemetry-go/compare/v1.2.0...v1.6.3)

---
updated-dependencies:
- dependency-name: go.opentelemetry.io/otel/trace
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ying-jeanne committed:
put the default logger format and feature toggle to fallback to old logger (#47584)

fridgepoet committed:
Fix wildcard on branches-ignore backport (#47679)

Fixes https://github.com/grafana/code-coverage/issues/11

dsotirakis committed:
Update grabpl (#47643)

ArturWierzbicki committed:
Previews: track oss adoption

gillesdemey commented on public/app/features/alerting/unified/utils/rulerClient.ts:
```diff
@@ -157,65 +141,105 @@ export function getRulerClient(rulerConfig: RulerDataSourceConfig): RulerClient
-      return { uid: newUid, ruleSourceName: 'grafana' };
-    } else {
-      throw new Error('Failed to fetch created rule.');
+    // TODO figure out how to get the newly created UID :(
```
**question:** sadly we don't have the means to find the UID, we just grabbed the UID of the only rule in the created group previously but we can no longer do that here. And suggestions or ideas more than welcome!

vtorosyan committed:
Reload permissions cache when adding a datasource (#47658)

gillesdemey commented:
> I wonder if we also should unify the way we allow to edit rule groups?

Yes absolutely, I was initially thinking of including it in the PR but it's grown quite large and I think we should tackle it incrementally.

renovate[bot] committed:
Update Storybook updates to v6.4.21 (#47523)

* Update Storybook updates to v6.4.21

* chore(yarn): update yarnrc storybook versions

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Jack Westbrook <jack.westbrook@gmail.com>

gillesdemey committed:
PR feedback

konrad147 committed:
Add permission checking to alert rules in the folders view (#47686)

dsotirakis committed:
Remove brackets (#47691)

eleijonmarck committed:
fix: bug where disabled didnt disable the use of service account (#47688)

jackw committed:
Chore: Bump moment to 2.29.2 (#47680)

* chore: bump moment to 2.29.2

* chore(yarn): clean up remaining moment nested deps

renovate[bot] committed:
Update dependency react-select-event to v5.5.0 (#47308)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency @betterer/betterer to v5.2.1 (#47700)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

zserge committed:
Chore: Remove bus.Bus field (#47695)

* Chore: Remove bus.Bus field

* fix integration test

kaydelaney committed:
Nav: Icon clickable area fills all available space (#47334)

Closes #47255

andresmgot committed:
AzureMonitor: Include macros and template variables as suggestions (#47689)

gillesdemey committed:
chore: clean up TODOs

TsotosA committed:
Explore: no data returned (#46987)

* Explore: no data message

display no data text when query executed succesfully but no data was
returned

* Explore: 'no-data' display logic and styling fix

* pipeline trigger

* Revert "pipeline trigger"

This reverts commit f08d246ba767e66344ad252577523f391832ec18.

* Add more parameters for when to show no data

* Change check to use frame length

Co-authored-by: Kristina Durivage <kristina.durivage@grafana.com>

connorlindsey committed:
NodeGraph: Detect dataframes more accurately based on fields (#47213)

* NodeGraph: Detect dataframes more accurately based on fields

* Make get fields case insensitive

* Update node graph docs

konrad147 committed:
Alerting: FGAC for alert rule view and edit page (#47441)

* Add permission check to the find route, add query not accessible warning for the view page

* Hide grafana or mimir rule buttons depending on user's permissions

* Add grafana and cloud read rules checking on the alert rules list view

* Improve missing data source handling, refactor edit and remove permissions handling

* Add tests for rule edit permissions

* PR feedback

Jguer committed:
AccessControl: fix value copying in method access and add LogID (#47694)

* AccessControl: fix value copying in method access and add LogID

Co-authored-by: Gabriel MABILLE <gamab@users.noreply.github.com>

* Lint.

Co-authored-by: Gabriel MABILLE <gamab@users.noreply.github.com>
Co-authored-by: gamab <gabi.mabs@gmail.com>

leeoniya committed:
TimeSeries & BarChart: refactor stacking (#47373)

Co-authored-by: Zoltán Bedi <zoltan.bedi@gmail.com>
Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>
Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

ying-jeanne committed:
fix terminal format (#47718)

IevaVasiljeva committed:
Access control: service account role check (#47710)

* forbid setting role higher than user's role

* change response code

* can assign API key permissions to non-admin users

* add: assign viewer role directly upon creation

* refactor: add AddSATcommand infavor of AddAPIkey

* refactor: frontend fixes for ServiceAccountToken

Co-authored-by: eleijonmarck <eric.leijonmarck@gmail.com>

eleijonmarck committed:
SAML: fix pkcs8 doc and refactoring SAML docs (#47337)

* refactoring: saml docs

* refactoring: relreferences

* adding error as presented in issue

* refactor: update sentence

* refactor: more relref fixes that was missing

* refactor: more relref fixes that was missing

* Apply suggestions from code review

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* refactor: move saml with okta

* fix: spell and small corrections

* add: enterprise tag

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

yuri-tceretian committed:
Alerting: Support OK option for Error state (#47670)

* support OK state for Error

FZambia committed:
Live: collect more usage stats – enabled, ha enabled, num nodes, channels, subs (#47378)

santihernandezc committed:
Apply templating on alert notifications on OK state (#47355)

* OK notifications using previous evaluation data

* copy rule.EvalMatches to avoid changes to the underlying array

* test cases added/modified

* delete trailing newline

* fix double newline in go import

* add change to the changelog

* specify that this only affects legacy alerting (changelog)

* use current eval data instead of prev eval data

* create evalMatch just once

* code comments, renamings, getTemplateMatches() function

* changelog and docs updated

JohnnyQQQQ committed:
Alerting: Provisioning API - Contact points (#47197)

santihernandezc committed:
notification template docs updated (#47744)

alexweav committed:
Alerting: Remove mis-behaving fake and fix masked test failure in AM config API (#47747)

* Remove misbehaving fake

* Fix bug and inject logger

ryantxu committed:
Heatmap: hover legend regardless of tooltip state (#47600)

ivanahuckova committed:
Query history: Add migration endpoint (#47551)

* Add endpoint for migration

* Check for createdAt

* Query history: Remove returning of dtos

* Query history: Fix CreatedAt

* Refactor based on suggestions

* Insert into table in batches

torkelo committed:
Panels: Use the No value option when showing the no data message (#47675)

* Performance: Create separate div for portal root to limit reflow -> recalc style

* refactoring

* NoData: Use field config No value option as text when display no data message

* Use PanelDataErrorView in TabelPanel

* Add PanelDataErrorView to PieChart

* Updated

renovate[bot] committed:
Update dependency @betterer/cli to v5.3.0 (#47703)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

ivanahuckova committed:
Loki: Use single string expr as a state for the visual editor (#47566)

* Loki: Use expr as state for visual editor

* Loki: Use query with line filter as default for visual editor

* Refactor based on feedback

* fix background for query text row

Co-authored-by: Torkel Ödegaard <torkel@grafana.com>

renovate[bot] committed:
Update dependency @betterer/betterer to v5.3.0 (#47757)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency @betterer/regexp to v5.3.0 (#47759)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

papagian committed:
API: enable proxying datasource calls using the datasource UID (#47634)

* Introduce additional routes

torkelo committed:
QueryEditorRow: Show query errors next to query in a consistent way across Grafana (#47613)

* Show query errors under each query row

* Testing a more plain error box

* Font size

* Make it green

* Nit UI

* Slight simplification of condition

* New design

* Update

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>

IevaVasiljeva committed:
Access control: expose SA frontend to users with the right permissions (#47727)

* expose frontend to users with permissions

* cover the ui endpoints

* fix permissions

gillesdemey committed:
Revert to a2aeea5

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=47527) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=47527) before we can accept your contribution.<br/>**6** out of **7** committers have signed the CLA.<br/><br/>:white_check_mark: ivanahuckova<br/>:white_check_mark: IevaVasiljeva<br/>:white_check_mark: torkelo<br/>:white_check_mark: ryantxu<br/>:white_check_mark: papagian<br/>:white_check_mark: gillesdemey<br/>:x: renovate[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=47527) it.</sub>

gillesdemey committed:
Merge branch 'main' into feat/grafana-managed-group-names

gillesdemey committed:
fix test
